backend/bin/deal-observer-backend.js

```diff
@@ -128,12 +129,13 @@ const sparkApiSubmitDealsLoop = async (pgPool, { sparkApiBaseUrl, sparkApiToken,
 
 export const pieceIndexerLoop = async (makeRpcRequest, getDealPayloadCid, pgPool) => {
   const LOOP_NAME = 'Piece Indexer'
+  const payloadsCache = new NodeCache()
```

Comment:
```suggestion
  const payloadsCache = new Map()
```

It looks like we're not using any node-catch features, so we can use a plain map instead